tests/language/generation/creator.ts

```diff
@@ -30,8 +30,13 @@ const createGenerationTest = async (parentDirectory: URI, inputUris: URI[]): Pro
     const actualOutputRoot = URI.file(path.join(parentDirectory.fsPath, 'generated'));
     const expectedOutputFiles = readExpectedOutputFiles(expectedOutputRoot, actualOutputRoot);
     let runUntil: Location | undefined;
+    // First read all stubs, then read all the other files; This should avoid broken references
+    const sortedInputUris = inputUris
```

Comment:
@lars-reimann The solution of calling `loadAllDocuments` before the call to `getErrors(services, code)` broke with duplicate definitions of segments and functions.
I temporarily used the old solution of sorting the files by extension. This could break, if definitions are needed from a file which is not an `.sdsstub` file.
Should I create a function similar to `getErrors(services, code)` below that supports working on a `LangiumDocument` instead of raw code?

Reply:
> Should I create a function similar to getErrors(services, code) below that supports working on a LangiumDocument instead of raw code?

That seems reasonable. It could also accept a URI instead of the document.